### PR TITLE
mds 3725 New Document management tab

### PR DESCRIPTION
**Pull request by JustinMacaulay**
Adds the new manage documents tab to the NoW application process as well as re-structuring the look of all the document tables.

### CODE REVIEW

File: services/core-web/src/components/noticeOfWork/applications/PermitPackage.js [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+import {
+  createNoticeOfWorkApplicationReview,
+  fetchNoticeOfWorkApplicationReviews,
+  deleteNoticeOfWorkApplicationReview,
+  updateNoticeOfWorkApplicationReview,
+  deleteNoticeOfWorkApplicationDocument,
+  setNoticeOfWorkApplicationDocumentDownloadState,
+  updateNoticeOfWorkApplication,
+  fetchImportedNoticeOfWorkApplication,
+} from "@common/actionCreators/noticeOfWorkActionCreator";
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/PermitPackage.js:16]: has this file been linted correctly? These functions should be listed in PropTypes.

File: services/core-web/src/components/noticeOfWork/applications/PermitPackage.js [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+  getNOWProgress,
+} from "@common/selectors/noticeOfWorkSelectors";
+import { EDIT_OUTLINE_VIOLET, EDIT_OUTLINE } from "@/constants/assets";
+import * as Permission from "@/constants/permissions";
+import { getDropdownNoticeOfWorkApplicationReviewTypeOptions } from "@common/selectors/staticContentSelectors";
+import NOWActionWrapper from "@/components/noticeOfWork/NOWActionWrapper";
+
+/**
+ * @constant ReviewNOWApplication renders edit/view for the NoW Application review step
+ */
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/PermitPackage.js:29]: Update Comment

File: services/core-web/src/components/noticeOfWork/applications/PreviousAmendmentDocuments.js [hunk 3]
```diff
@@ -7,13 +7,11 @@ import { renderConfig } from "@/components/common/config";
 
 const propTypes = {
   previousAmendmentDocuments: PropTypes.arrayOf(PropTypes.any),
-  showPreambleFileMetadata: PropTypes.bool,
   editPreambleFileMetadata: PropTypes.bool,
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/PreviousAmendmentDocuments.js:10]: was this unused?

File: services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+import { getDropdownInspectors } from "@common/selectors/partiesSelectors";
+import CustomPropTypes from "@/customPropTypes";
+import * as FORM from "@/constants/forms";
+import NOWSideMenu from "@/components/noticeOfWork/applications/NOWSideMenu";
+import NOWTabHeader from "@/components/noticeOfWork/applications/NOWTabHeader";
+import NOWApplicationManageDocuments from "@/components/noticeOfWork/applications/manageDocuments/NOWApplicationManageDocuments";
+
+/**
+ * @class ManageDocumentsTab- contains all information relating to the Securities/Bond tracking on a Notice of Work Application.
+ */
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js:26]: update comment

File: services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+export class ManageDocumentsTab extends Component {
+  state = {
+    isInspectorsLoaded: true,
+  };
+
+  componentDidMount = () =>
+    this.props.fetchNoticeOfWorkApplicationReviews(this.props.noticeOfWork.now_application_guid);
+
+  handleUpdateInspectors = (values, finalAction) => {
+    this.setState({ isInspectorsLoaded: false });
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js]: Is this needed on the manage documents tab? Where are inspectors being updated on this tab?

File: services/core-web/src/components/noticeOfWork/applications/manageDocuments/NOWApplicationManageDocuments.js [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+import PropTypes from "prop-types";
+import CustomPropTypes from "@/customPropTypes";
+import FinalPermitDocuments from "@/components/noticeOfWork/applications/FinalPermitDocuments";
+import NOWDocuments from "@/components/noticeOfWork/applications//NOWDocuments";
+import ScrollContentWrapper from "@/components/noticeOfWork/applications/ScrollContentWrapper";
+import NOWSubmissionDocuments from "@/components/noticeOfWork/applications//NOWSubmissionDocuments";
+
+/**
+ * @class NOWApplicationManageDocuments- contains all information relating to the Administrative work on a Notice of Work Application
+ */
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/manageDocuments/NOWApplicationManageDocuments.js:11]: update comment

File: services/core-web/src/styles/settings/themeOverride.scss [hunk 7]
```diff
@@ -637,3 +637,8 @@ select {
 .ant-descriptions-item-label {
   font-weight: 600;
 }
+
+.ant-tooltip-disabled-compatible-wrapper {
+  width: inherit;
```
Comment (mmumby) [inline services/core-web/src/styles/settings/themeOverride.scss]: Was this used in the end? If not, remove

File: services/core-web/common/actionCreators/noticeOfWorkActionCreator.js [hunk 8]
```diff
@@ -378,12 +378,15 @@ export const deleteNoticeOfWorkApplicationReview = (applicationGuid, application
...
+  dispatch
+) => {
   dispatch(showLoading("modal"));
   dispatch(request(reducerTypes.ADD_DOCUMENT_TO_NOTICE_OF_WORK));
   return CustomAxios()
     .put(
-      ENVIRONMENT.apiUrl + API.NOTICE_OF_WORK_DOCUMENT(now_application_guid),
+      `${ENVIRONMENT.apiUrl +
+        API.NOTICE_OF_WORK_DOCUMENT(now_application_guid)}/${mineDocumentGuid}`,
       payload,
```
Comment (mmumby) [inline services/core-web/common/actionCreators/noticeOfWorkActionCreator.js:390]: The const 'NOTICE_OF_WORK_DOCUMENT' should be updated to include the new param in the API.js instead.

```
export const NOTICE_OF_WORK_DOCUMENT = (now_document_guid, mineDocumentGuid) =>
  `/now-applications/${now_document_guid}/document/${mineDocumentGuid}`;
  ```

File: services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js [hunk 9]
```diff
@@ -39,33 +42,65 @@ const propTypes = {
...
+  isRefConDocuments: false,
+  isPackageModal: false,
+  showDescription: false,
 };
 
+export const RenderNowDocumentsTableExpandIcon = (rowProps) => (
+  <div>
+    {rowProps.expanded ? (
+      <Tooltip title="Click to hide amendment history." placement="right" mouseEnterDelay={1}>
+        <MinusSquareFilled className="icon-lg--lightgrey" />
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js:77]: Update tooltip to mention documents and not amendments

File: services/core-web/src/components/noticeOfWork/applications/PermitPackage.js [hunk 10]
```diff
@@ -0,0 +1,287 @@
...
+    const poll = (resolve) => {
+      if (conditionFunction()) resolve();
+      else setTimeout(() => poll(resolve), 400);
+    };
+
+    return new Promise(poll);
+  };
+
+  downloadDocumentPackage = () => {
+    const docURLS = [];
```
Comment (mmumby) [inline services/core-web/src/components/noticeOfWork/applications/PermitPackage.js:112]: Can you confirm that the download functionality and cancel download still works? I didn't test this feature.

**Review by mmumby — CHANGES_REQUESTED** [review]
Nice work, there is a lot to follow here. 
I have a few questions/comments
and can you add snapshot tests for the new components added?

File: services/core-web/src/components/noticeOfWork/applications/PermitPackage.js (hunk 1 above)
Comment (JustinMacaulay) [reply services/core-web/src/components/noticeOfWork/applications/PermitPackage.js:16]: I did lint this and got no errors but I will take another look at it

Comment (JustinMacaulay) [reply services/core-web/src/components/noticeOfWork/applications/PermitPackage.js:16]: I removed the unused ones but they were not throwing a linting error

File: services/core-web/src/components/noticeOfWork/applications/PermitPackage.js (hunk 2 above)
Comment (JustinMacaulay) [reply services/core-web/src/components/noticeOfWork/applications/PermitPackage.js:29]: updated

File: services/core-web/src/components/noticeOfWork/applications/PreviousAmendmentDocuments.js (hunk 3 above)
Comment (JustinMacaulay) [reply services/core-web/src/components/noticeOfWork/applications/PreviousAmendmentDocuments.js:10]: it was and linting was unhappy

File: services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js (hunk 4 above)
Comment (JustinMacaulay) [reply services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js:26]: updated

File: services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js (hunk 5 above)
Comment (JustinMacaulay) [reply services/core-web/src/components/noticeOfWork/applications/manageDocuments/ManageDocumentsTab.js]: I think this was a copy paste by mehkti cause he used the administrative tab as his base. I am removing it.

File: services/core-web/src/components/noticeOfWork/applications/manageDocuments/NOWApplicationManageDocuments.js (hunk 6 above)
Comment (JustinMacaulay) [reply services/core-web/src/components/noticeOfWork/applications/manageDocuments/NOWApplicationManageDocuments.js:11]: updated

File: services/core-web/src/styles/settings/themeOverride.scss (hunk 7 above)
Comment (JustinMacaulay) [reply services/core-web/src/styles/settings/themeOverride.scss]: removed

File: services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js (hunk 9 above)
Comment (JustinMacaulay) [reply services/core-web/src/components/noticeOfWork/applications/NOWDocuments.js:77]: updated

File: services/core-web/src/components/noticeOfWork/applications/PermitPackage.js (hunk 10 above)
Comment (JustinMacaulay) [reply services/core-web/src/components/noticeOfWork/applications/PermitPackage.js:112]: I have used it and as far as I can tell it works correctly.

File: services/core-web/common/actionCreators/noticeOfWorkActionCreator.js (hunk 8 above)
Comment (JustinMacaulay) [reply services/core-web/common/actionCreators/noticeOfWorkActionCreator.js:390]: this is used in multiple areas and not just for this. I dont see how this is bad or incorrect though.

Comment (mmumby) [reply services/core-web/common/actionCreators/noticeOfWorkActionCreator.js:390]: It's not necessarily bad, we just have a pattern for our API routes. If this is being used in multiple places that do not need `mineDocumentGuid` i guess we can keep it like this.

**Review by mmumby — DISMISSED** [review]
Looks good, Thanks for adding the tests! 🎉

**Review by mmumby — APPROVED** [review]

**Pull request merged**